Updated Dart sdk to v4.0.0 Catnip (#365)
configcat/docs: website/docs/sdk-reference/dart.mdx

);
```

:::caution
It is important to provide an argument for the `defaultValue` parameter, specifically for the `T` generic type parameter,
that matches the type of the feature flag or setting you are evaluating. Please refer to the following table for the corresponding types.
:::

### Setting type mapping {#setting-type-mapping}

| Setting Kind   | Type parameter `T` |
| -------------- |--------------------|
| On/Off Toggle  | `bool`/`bool?`     |
| Text           | `String`/`String?` |
| Whole Number   | `int`/`int?`       |
| Decimal Number | `double`/`double?` |

> In addition to the types mentioned above, you also have the option to provide `dynamic` for the type parameter regardless of the setting kind.

Reviewer: ```suggestion
In addition to the types mentioned above, you also have the option to provide `object`, `object?` or `dynamic` for the type parameter regardless of the setting kind.
```

);
```

:::caution
It is important to provide an argument for the `defaultValue` parameter, specifically for the `T` generic type parameter,
that matches the type of the feature flag or setting you are evaluating. Please refer to [this table](#setting-type-mapping) for the corresponding types.
:::

The `details` result contains the following information:

| Field                       | Type                                 | Description                                                                               |
|-----------------------------|--------------------------------------|-------------------------------------------------------------------------------------------|
| `value`                     | `bool` / `String` / `int` / `double` | The evaluated value of the feature flag or setting.                                       |
| `key`                       | `String`                             | The key of the evaluated feature flag or setting.                                         |

> | `isDefaultValue`            | `bool`                               | True when the default value passed to getValueDetails() is returned due to an error.      |

Reviewer: ```suggestion
| `isDefaultValue`            | `bool`                               | True when the default value passed to `getValueDetails()` is returned due to an error.      |
```

| `error`                     | `String?`                            | In case of an error, this field contains the error message.                               |
| `user`                      | `ConfigCatUser?`                     | The User Object that was used for evaluation.                                             |
| `matchedPercentageOption`   | `PercentageOption?`                  | If the evaluation was based on a percentage rule, this field contains that specific rule. |

> | `matchedTargetingRule`      | `TargetingRule?`                     | If the evaluation was based on a Targeting Rule, this field contains that specific rule.  |

Reviewer: ```suggestion
| `matchedPercentageOption`   | `PercentageOption?`                  | The Percentage Option (if any) that was used to select the evaluated value. |
| `matchedTargetingRule`      | `TargetingRule?`                     | The Targeting Rule (if any) that matched during the evaluation and was used to return the evaluated value.  |
```

* accept `String` values containing a properly formatted, valid semver value,
* all other values are considered invalid (a warning will be logged and the currently evaluated targeting rule will be skipped).

**Number-based comparators** (=, &lt;, &gt;=, etc.)
* accept `double` values and all other numeric values which can safely be converted to `double`,
* accept `String` values containing a properly formatted, valid `double` value,
* all other values are considered invalid (a warning will be logged and the currently evaluated targeting rule will be skipped).

**Date time-based comparators** (BEFORE / AFTER)
* accept `DateTime` values, which are automatically converted to a second-based Unix timestamp,
* accept `double` values representing a second-based Unix timestamp and all other numeric values which can safely be converted to `Double`,
* accept `String` values containing a properly formatted, valid `double` value,
* all other values are considered invalid (a warning will be logged and the currently evaluated targeting rule will be skipped).

**String array-based comparators** (ARRAY CONTAINS ANY OF / ARRAY NOT CONTAINS ANY OF)

> * accept arrays and list of `String`,
> * accept `String` values containing a valid JSON string which can be deserialized to an array of `String`,

Reviewer: https://dart.dev/language/collections#lists

> Perhaps the most common collection in nearly every programming language is the array, or ordered group of objects. In Dart, arrays are [List](https://api.dart.dev/stable/dart-core/List-class.html) objects, so most people just call them lists.

```suggestion
* accept lists or sets of `String`,
* accept `String` values containing a valid JSON string which can be deserialized to a list of `String`,
```
